fix: improve Sort By dropdown styling consistency

mrkeshav-05 opened the pull request:
- Remove blue outline and inconsistent hover backgrounds
- Add proper focus/active states styles to match the design system
- Improve light/dark mode theme integration
- Fix trigger background blending issues


Fixes #1372

Before Changes:
| Light Mode | Dark Mode |
|--------|--------|
| <img width="348" height="223" alt="image" src="https://github.com/user-attachments/assets/62801042-4b48-401f-9fd8-80fff72f442d" /> | <img width="327" height="225" alt="image" src="https://github.com/user-attachments/assets/f44b1ef6-b21a-43bd-acc6-5039a265a9b5" /> |
| <img width="351" height="274" alt="image" src="https://github.com/user-attachments/assets/598593f5-375d-4d00-a808-3f41b42d1378" /> | <img width="310" height="263" alt="image" src="https://github.com/user-attachments/assets/c4f2ae4e-f5f5-4f93-a38e-6d1a18d39c0f" /> | 

After Changes:
| Light Mode | Dark Mode |
|--------|--------|
| <img width="309" height="230" alt="image" src="https://github.com/user-attachments/assets/f165c8ef-84fa-4d65-a97d-1759dc249863" /> | <img width="355" height="214" alt="image" src="https://github.com/user-attachments/assets/24b08842-9fc5-4fbb-b121-9a4fce2507ec" /> |
| <img width="351" height="328" alt="image" src="https://github.com/user-attachments/assets/6676036b-b8cf-4561-add3-6678b19c2873" /> | <img width="339" height="329" alt="image" src="https://github.com/user-attachments/assets/a9630e6e-90b3-4c89-aa53-863135b1d628" /> |

kasya commented on frontend/src/components/SortBy.tsx:
Can we make sure the padding on right and left side match? Right now it's way bigger on the left:

<img width="282" height="77" alt="Image" src="https://github.com/user-attachments/assets/d84d3443-2baf-4c23-b79f-ac6b5148b68e" />

<img width="262" height="85" alt="Image" src="https://github.com/user-attachments/assets/fa7dd8f6-b38d-4344-9729-6de100d43a34" />

kasya reviewed (changes requested):
Thanks @mrkeshav-05 for working on this!
Left some comments. 

Also, please run `make check` and commit suggested changes - this is required by contributing guidelines in OWASP Nest project.

mrkeshav-05 commented:
@kasya Thank you for the review!

I've addressed the feedback:
- Ran `make check` - all checks passed successfully
- Fixed the padding imbalance issue - added balanced `px-3` to the container
- Added "hover:bg-transparent dark:hover:bg-transparent"  for Light and Dark mode


screenshot of the improved style:
| Light Mode | Dark Mode |
|--------|--------|
| <img width="345" height="334" alt="image" src="https://github.com/user-attachments/assets/f23e993c-9851-425a-bad9-7f9c60ca518e" /> | <img width="406" height="331" alt="image" src="https://github.com/user-attachments/assets/c4264e31-bf53-488e-9722-bc2dade1311c" /> | 

Code is ready for review.
